src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala

```diff
@@ -14,7 +14,7 @@ abstract class WorkerMain[S](stdin: InputStream = System.in, stdout: PrintStream
 
   protected[this] def init(args: Option[Array[String]]): S
 
-  protected[this] def work(ctx: S, args: Array[String], out: PrintStream, workDir: Path): Unit
+  protected[this] def work(ctx: S, args: Array[String], out: PrintStream, workDir: Path, verbosity: Int): Unit
```

Comment:
Since we're getting the arguments and verbosity directly from the work request, should we just pass that to the worker directly instead of copying so many of its fields?

Reply:
I'm not sure we should because this does create a nice interface all the workers must adhere to. It does sometimes mean updates when the protocol changes. Good news is that tends to happen rather infrequently.

Reply:
Yeah, I can see how this forces worker implementers to think about certain things. Nevermind, then.